### ISSUE TITLE
{{sub}} seems to accept array instead of N arguments

### ISSUE THREAD

**Issue by MichalBryxi** [RobbieTheWagner/ember-math-helpers#1623]
The [documentation](https://robbiethewagner.github.io/ember-math-helpers/docs/api/helpers/sub) says:

```
{{sub a b}}
```

But if I try that I will get:

![Screenshot 2024-06-20 at 13 11 22](https://github.com/RobbieTheWagner/ember-math-helpers/assets/847473/351ecf98-23c9-42ad-8345-88be7248cd52)

Moreover the documentation also defines `sub` as:

```
sub(numbers: number[]): number
```

Which seems to align with the idea of passing down an array? So the example code should be:

```
{{sub (array a b)}}
```

That being said, I find that a bit weird API and would hope for `{{sub a b}}` to simply work.

**Comment by MichalBryxi** [comment]
My best guess would be that [this commit](https://github.com/RobbieTheWagner/ember-math-helpers/pull/1163/files#diff-485a79d51bb797e9b8b2b5b4a75ecd82d1837947c792b416bc1da4dd57df8f4c) is the culprit?

**Comment by RobbieTheWagner** [comment]
@MichalBryxi seems like maybe just the types are wrong?

**Comment by MichalBryxi** [comment]
> @MichalBryxi seems like maybe just the types are wrong?

I'm not sure what that means? Should I update some package? The types are afaik `number` as per the example: `{{sub 2 1}}`.

**Comment by RobbieTheWagner** [comment]
@MichalBryxi I am saying the types are likely wrong. The helper has not changed and works as documented, but TS/glint is what you seem to be running into issues with. I myself have not experienced the same issue, but that is likely where the issue lies.

**Comment by RobbieTheWagner** [comment]
Are you importing the registry and adding it to your local glint registry? It seems to be working fine for me.

**Comment by MichalBryxi** [comment]
Oooh, in the end this might be a tiny bit different issue:

1) Current [configuration section of the docs for `master`](https://robbiethewagner.github.io/ember-math-helpers/versions/master/docs/configuration) does not show the [glint / template imports](https://github.com/RobbieTheWagner/ember-math-helpers/blob/main/docs/docs/configuration.md?plain=1#L21-L47) part of the `configuration.md`.
2) The docs pages show only `master`, but the repo already uses `main` and there is not `master`.
3) The docs pages show as `lastest = v3.0.0`

So no, I did not add stuff to the registry as I did not see it in the docs :)

**Comment by MichalBryxi** [comment]
To document this for my future self: I forgot on which project I had this issue and can not find anything that would have said configuration now. But I bumped onto vaguely similar issue in #1673. Might be that I can't provide reproduction for this one anymore, for which I'm deeply sorry.

**Comment by MichalBryxi** [comment]
I think I have quite a solid reproduction showing that the `v4.0.0` version of `{{sum}}` does indeed _need_ `{{sum (array 1 2)}}` syntax and will crash with `TypeError: numbers.reduce is not a function` when called as `{{sum 1 2}}`.

I expanded my bugreport app [here](https://gitlab.com/michal-bryxi/bugreport/ember-math-helpers-bugreport/-/commit/52fd59867e0bbbcfc5d5e9b0972b2e153f02965b) and [here](https://gitlab.com/michal-bryxi/bugreport/ember-math-helpers-bugreport/-/commit/8b3b63c6fefcd504db819e94711ac7eb78c39648)

IIRC there was some shenanigans with `helpers` in Ember that based on [something something] you could either provide them with series of arguments or needed to provide an argument array. Let me poke discord if someone can put it into better words.

**Comment by MichalBryxi** [comment]
Yup, seems like it's somewhere on the line as I stated above: https://discord.com/channels/480462759797063690/1268557129791443048

> "a fix" could be to just export-default sub and delete the SubSignature
> there is no need to wrap things in helper anymore, unless you want to support prior to ember 4.5, without the polyfill

cc: @RobbieTheWagner
